app/src/main/java/de/westnordost/streetcomplete/quests/railway_crossing/AddRailwayCrossingBarrier.kt

```diff
@@ -1,21 +1,41 @@
 package de.westnordost.streetcomplete.quests.railway_crossing
 
+import de.westnordost.osmapi.map.data.BoundingBox
+import de.westnordost.osmapi.map.data.Element
 import de.westnordost.streetcomplete.R
-import de.westnordost.streetcomplete.data.osm.SimpleOverpassQuestType
+import de.westnordost.streetcomplete.data.osm.OsmElementQuestType
 import de.westnordost.streetcomplete.data.osm.changes.StringMapChangesBuilder
+import de.westnordost.streetcomplete.data.osm.download.MapDataWithGeometryHandler
 import de.westnordost.streetcomplete.data.osm.download.OverpassMapDataDao
+import de.westnordost.streetcomplete.data.osm.tql.OverpassQLUtil
+import de.westnordost.streetcomplete.quests.AbstractQuestAnswerFragment
 
-class AddRailwayCrossingBarrier(o: OverpassMapDataDao) : SimpleOverpassQuestType<String>(o) {
-
-    override val tagFilters = "nodes with railway=level_crossing and !crossing:barrier"
+class AddRailwayCrossingBarrier(private val overpassMapDataDao: OverpassMapDataDao) : OsmElementQuestType<String> {
     override val commitMessage = "Add type of barrier for railway crossing"
     override val icon = R.drawable.ic_quest_railway
 
     override fun getTitle(tags: Map<String, String>) = R.string.quest_railway_crossing_barrier_title
 
     override fun createForm() = AddRailwayCrossingBarrierForm()
 
+    override fun download(bbox: BoundingBox, handler: MapDataWithGeometryHandler): Boolean {
+        return overpassMapDataDao.getAndHandleQuota(getOverpassQuery(bbox), handler);
+    }
+
+    override fun isApplicableTo(element: Element): Boolean? = null
+
     override fun applyAnswerTo(answer: String, changes: StringMapChangesBuilder) {
         changes.add("crossing:barrier", answer)
     }
+
+    private fun getOverpassQuery(bbox: BoundingBox): String? {
```

Comment:
Here's a little nitpicking, no need to change it if you dont' want to: The method declares itself to return "maybe" a string (= a `String` or `null`) but looking at the code, it definitely always returns a `String`. So, you can remove the `?`.

This does not matter at the moment because `OverpassMapDataDao` is a Java class which thus doesn't know the difference between a `String` and `String?`.

Reply:
fixed!